File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] n) {
        for(int i=0;i<n.length;i++){
            while(n[i]>0 && n[i] <= n.length && n[n[i]-1]!=n[i]){
                int t=n[i];
                n[i]=n[n[i]-1];
                n[t-1]=t;
            }
        }
        for(int i=0;i<n.length;i++){
            if(i+1!=n[i]){
                return i+1;
            }
        }
        return n.length+1;
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, n: List[int]) -> int:
        a=[0]*(len(n)+1)
        for i in n:
            if(i>0 and i<=len(n)):
                a[i-1]=1
        #print(a)
        for i in range(len(a)):
            if(a[i]==0):
                return i+1
        return -1
        